### DISCUSSION TITLE
Mentions command panel

### DISCUSSION THREAD

**Discussion by aytunch** [flyerhq/flutter_chat_ui#656] [Ideas & Requests]
Especially in group chats, many modern chat apps have mentioning feature.
Let's say we have a list of `User`s for our group chat. When the user types `@` in to the chat text field, we should show the list of users in a popup.
And when the user keeps typing letters, we should filter the users according to the filter text.
If the user chooses one of the suggested users to mention, that user name should have a different style in the text field, telling the user that they will be mentioned.

I know there are separate packages for the UI which do these. If you think this is a good addition and want more info I can supply more resources.

Thanks for this great package.

The screen shots are from desktop apps of Discord and Slack for reference.

<img width="767" alt="Screenshot 2022-12-23 at 22 43 34" src="https://user-images.githubusercontent.com/6442915/209399517-3b484bf6-a3a7-47bc-9baf-d5bd5bfe5f27.png">

<img width="1458" alt="Screenshot 2022-12-23 at 22 44 01" src="https://user-images.githubusercontent.com/6442915/209399511-1f6eddc7-7538-4454-8ecf-d363ac0c15b9.png">

**Comment by demchenkoalex** [comment]
Hey, what separate packages you have in mind?

**Comment by leeyisoft** [comment]
I'm going to add a group chat function to my open source project, and I need a similar effect

I know https://github.com/fluttercandies/extended_text AtText class is part of the function, but not all, are less than perfect

**Comment by demchenkoalex** [comment]
I wish there would be something like this for flutter https://github.com/Expensify/react-native-live-markdown
